src/border_agent/border_agent.cpp

```diff
@@ -252,8 +251,14 @@ void BorderAgent::PublishMeshCopService(void)
         }
         else
         {
-            activeTimestamp = htobe64(activeDataset.mActiveTimestamp);
-            txtList.emplace_back("at", reinterpret_cast<uint8_t *>(&activeTimestamp), sizeof(activeTimestamp));
+            uint64_t activeTimestampValue;
+
+            activeTimestampValue = (activeDataset.mActiveTimestamp.mSeconds << 16) |
```

Comment:
nit: we can elide the bit operations if `otTimestamp` is defined as an union with a `uint64_t` value to get the full timesstamp.

Reply:
ditto~